src/Nethermind/Nethermind.Optimism/CL/P2P/OptimismCLP2P.cs

```diff
@@ -175,26 +276,62 @@ private bool TryValidateAndDecodePayload(byte[] msg, [MaybeNullWhen(false)] out
         return true;
     }
 
+    private async Task<ISession?> TryDialPeer(Multiaddress remotePeer, CancellationToken token)
+    {
+        try
+        {
+            CancellationTokenSource cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
+            cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(5));
+            return await _localPeer!.DialAsync(remotePeer, cancellationTokenSource.Token);
+        }
+        catch (Exception e)
+        {
+            if (_logger.IsTrace) _logger.Trace($"Unable to Dial peer {remotePeer}. Exception: {e.Message}");
+            return null;
+        }
+    }
+
     public async Task Run(CancellationToken token)
     {
         if (_logger.IsInfo) _logger.Info("Starting Optimism CL P2P");
 
         IPeerFactory peerFactory = _serviceProvider.GetService<IPeerFactory>()!;
-        string address = $"/ip4/{_config.P2PHost}/tcp/{_config.P2PPort}";
+        string hostIp = _config.ClP2PHost ?? _externalIp.ToString();
+        string address = $"/ip4/{hostIp}/tcp/{_config.ClP2PPort}";
         _localPeer = peerFactory.Create(new Identity());
 
         _router = _serviceProvider.GetService<PubsubRouter>()!;
-
         _blocksV2Topic = _router.GetTopic(_blocksV2TopicId);
         _blocksV2Topic.OnMessage += msg => OnMessage(msg, token);
+        try
+        {
+            await _localPeer.StartListenAsync([address], token);
+            await _router.StartAsync(_localPeer, token);
+
+            _peerStore = _serviceProvider.GetService<PeerStore>()!;
+            foreach (var multiaddress in _staticPeerList)
```

Comment:
Same as above, should we run this in parallel?

Reply:
This is part of initialization code. No need for parallelization